### PR TITLE
Minor change: Reduce unnecessary judgment

### PR DESCRIPTION
**Pull request by ChenglongGao327**
This function is just for constructing mode.Service[], MeshExternal should be the same as other parameters, and can be used after passing it, without needing to judge again

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/serviceregistry/serviceentry/conversion.go [hunk 1]
```diff
@@ -191,19 +192,20 @@ func convertServices(cfg config.Config) []*model.Service {
...
 	}
+	meshExternal := serviceEntry.Location == networking.ServiceEntry_MESH_EXTERNAL
 
-	return buildServices(hostAddresses, cfg.Namespace, svcPorts, serviceEntry.Location, resolution,
+	return buildServices(hostAddresses, cfg.Namespace, svcPorts, meshExternal, resolution,
 		exportTo, labelSelectors, serviceEntry.SubjectAltNames, creationTime, cfg.Labels)
 }
 
-func buildServices(hostAddresses []*HostAddress, namespace string, ports model.PortList, location networking.ServiceEntry_Location,
+func buildServices(hostAddresses []*HostAddress, namespace string, ports model.PortList, meshExternal bool,
```
Comment (ramaraochavali) [inline pilot/pkg/serviceregistry/serviceentry/conversion.go:201]: What is the purpose of this change? Passing location seems better so that `buildServices` can use it for other purpose if needed?

**Comment by howardjohn** [comment]
Please indicate what is being improved in your PRs in the PR description in the future to help us review better. Thanks!

File: pilot/pkg/serviceregistry/serviceentry/conversion.go (hunk 1 above)
Comment (ChenglongGao327) [reply pilot/pkg/serviceregistry/serviceentry/conversion.go:201]: > What is the purpose of this change? Passing location seems better so that `buildServices` can use it for other purpose if needed?

I think this function is just for constructing mode.Service[], MeshExternal should be the same as other parameters, and can be used after passing it, without needing to judge again. I haven't thought of other scenarios that Locality can use yet.

**Comment by ChenglongGao327** [comment]
> Please indicate what is being improved in your PRs in the PR description in the future to help us review better. Thanks!

OK, i will add the detailed description for PR in the future

File: pilot/pkg/serviceregistry/serviceentry/conversion.go (hunk 1 above)
Comment (ramaraochavali) [reply pilot/pkg/serviceregistry/serviceentry/conversion.go:201]: I understand it is not used now but can be in future and I see no benefit of judging it before hand.
